sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_quickpulse/_utils.py

```diff
@@ -258,15 +242,217 @@ def _check_metric_filters(metric_infos: List[DerivedMetricInfo], data: _Telemetr
+        else:
+            val = getattr(data, field.name, None)
+            if val is not None:
+                if _field_string_compare(str(val), filter.comparand, filter.predicate):
+                    return True
+    return False
+
+
+def _check_custom_dim_field_filter(filter: FilterInfo, custom_dimensions: Dict[str, str]) -> bool:
+    field = filter.field_name.replace("CustomDimensions.", "")
+    value = custom_dimensions.get(field)
+    if value is not None:
+        return _field_string_compare(str(value), filter.comparand, filter.predicate)
+    return False
+
+
+def _field_string_compare(value: str, comparand: str, predicate: str) -> bool:
+    if predicate == PredicateType.EQUAL:
+        return value == comparand
+    if predicate == PredicateType.NOT_EQUAL:
+        return value != comparand
+    if predicate == PredicateType.CONTAINS:
+        return comparand.lower() in value.lower()
+    if predicate == PredicateType.DOES_NOT_CONTAIN:
+        return comparand.lower() not in value.lower()
+    return False
+
+
+# Validation
+
+
+def _validate_derived_metric_info(metric_info: DerivedMetricInfo) -> bool:
+    # Validate telemetry type
+    try:
+        telemetry_type = TelemetryType(metric_info.telemetry_type)
+    except Exception:  # pylint: disable=broad-except,invalid-name
+        return False
+    # Only REQUEST, DEPENDENCY, EXCEPTION, TRACE are supported
+    # No filtering options in UX for PERFORMANCE_COUNTERS
+    if telemetry_type not in (
+        TelemetryType.REQUEST,
+        TelemetryType.DEPENDENCY,
+        TelemetryType.EXCEPTION,
+        TelemetryType.TRACE,
+    ):
+        return False
+    # Check for CustomMetric projection
+    if metric_info.projection and metric_info.projection.startswith("CustomMetrics."):
+        return False
+    # Validate filters
+    for filter_group in metric_info.filter_groups:
+        for filter in filter_group.filters:
+            # Validate field names to telemetry type
+            # Validate predicate and comparands
+            if not _validate_filter_field_name(filter, telemetry_type) or not _validate_filter_predicate_and_comparand(
+                filter
+            ):
+                return False
+    return True
+
+
+# pylint: disable=R0911
+def _validate_filter_field_name(filter: FilterInfo, telemetry_type: TelemetryType) -> bool:
+    name = filter.field_name
+    if not name:
+        return False
+    if name.startswith("CustomMetrics"):
+        return False
+    if name.startswith("CustomDimensions") or name == "*":
+        return True
+    name = name.lower()
+    if telemetry_type == TelemetryType.DEPENDENCY:
+        if name not in _DEPENDENCY_DATA_FIELD_NAMES:
+            return False
+    elif telemetry_type == TelemetryType.REQUEST:
+        if name not in _REQUEST_DATA_FIELD_NAMES:
+            return False
+    elif telemetry_type == TelemetryType.EXCEPTION:
+        if name not in ("exception.message", "exception.stacktrace"):
```

Comment:
The UI has these names come through as Exception.Message & Exception.StackTrace
just making sure the casing doesn't cause any issues when it comes to filtering